meta-balena-common/recipes-containers/docker-disk/docker-disk.bb

```diff
@@ -55,6 +55,10 @@ do_compile () {
 	# mismatches
 	DOCKER=$(PATH="/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin" which docker)
 
+	# Force Go net module to use pure go resolver
+	# This prevents TLS handshake timeouts when net falls back to the cgo resolver
+	export GODEBUG=netdns=go
+
```

Comment:
@jakogut how is the error this fixes reproduced? I have never experienced this myself or seen this reported.
From the golang documentation it seems the go resolver is the default except for .local and mDNS where a cgo based resolver is used. Did you check whether forcing the go resolver affects local name resolution? As I read this (https://github.com/golang/go/commit/b711f5ad8f945a6c8e823cb1e7fc65eafdb3a3e4) it may well be the go resolver does not support mDNS.
One use case for mDNS is when you build agains a local devenv which is setup with a local domain. This will be the case for example to test hostapp extensions and supervisor containers locally.